src/wasm/extract.cpp

```diff
@@ -842,19 +855,19 @@ uint64_t extractor::seek_input_stream(stream::chunk_reader::type* chunk_source,
   return file.offset + file.size;
 }
 
-uint64_t extractor::copy_data(const stream::file_reader::pointer& source,
-                              const std::vector<file_output*>& outputs) {
+uint64_t extractor::copy_data(const stream::file& file, const stream::file_reader::pointer& source,
+                              file_output* output) {
   uint64_t output_size = 0;
   while (!source->eof()) {
     char buffer[8192 * 10];
     const auto buffer_size = std::streamsize(boost::size(buffer));
     const auto extracted_n = source->read(buffer, buffer_size).gcount();
+    wasm_debug("copy_data read, extracted_n=" << extracted_n);
+
     if (extracted_n > 0 || (source->eof() && !source->bad())) {
-      for (auto output : outputs) {
-        bool success = output->write(buffer, extracted_n);
-        if (!success) {
-          throw std::runtime_error("Error writing file \"" + output->path().string() + '"');
-        }
+      bool success = output->write(buffer, extracted_n, file.size);
+      if (!success) {
```

Comment:
Since you don't use that "success" variable anywhere below, I would recommend to simplify that code to:
     ` if (!output->write(buffer, extracted_n, file.size)) {`

Reply:
Valid point, although you can probably see this part was already present in the code, probably to make this part a bit more readable.
I agree that it can and probably should be simplified at one point or another, although a general refactor wasn't my goal when making this PR.